x/concentrated-liquidity/fees.go

```diff
@@ -264,15 +247,18 @@ func (k Keeper) collectFees(ctx sdk.Context, owner sdk.AccAddress, positionId ui
 func (k Keeper) GetClaimableFees(ctx sdk.Context, positionId uint64) (sdk.Coins, error) {
 	// Since this is a query, we don't want to modify the state and therefore use a cache context.
 	cacheCtx, _ := ctx.CacheContext()
+	return k.getClaimableFees(cacheCtx, positionId)
+}
 
+func (k Keeper) getClaimableFees(ctx sdk.Context, positionId uint64) (sdk.Coins, error) {
 	// Get the position with the given ID.
-	position, err := k.GetPosition(cacheCtx, positionId)
+	position, err := k.GetPosition(ctx, positionId)
```

Comment:
I do not like the way this was separated. Any get function should not have the ability to change state, even if its private. In the future, if someone sees this "getClaimableFees" private methods and attempts to use it in a new function, it would actually claim the rewards. We should make it so that there is no way for the getter to modify state, even if misused.

Reply:
Modifying state is intentional since it is used by collectFees as well. I can change the name to reflect that this is not a getter

Reply:
If the name changes then I am good with that!

Reply:
+1 on this change

Reply:
Changed to `prepareClaimableFees` and added godoc